### PR TITLE
Add custom settings uploader

### PR DESCRIPTION
**Pull request by davidmreed**
# Critical Changes

# Changes

- CumulusCI now includes a task, `load_custom_settings`, to upload YAML-defined Custom Settings into a target org.

# Issues Closed

### CODE REVIEW

File: cumulusci/cumulusci.yml [hunk 1]
```diff
@@ -409,6 +409,9 @@ tasks:
         options:
             mapping: 'datasets/mapping.yml'
             sql_path: 'datasets/sample.sql'
+    upload_custom_settings:
+        description: Upload Custom Settings specified in a YAML file to the target org
+        class_path: cumulusci.tasks.salesforce.UploadCustomSettings
```
Comment (davisagli) [inline cumulusci/cumulusci.yml]: Please set a `group` for each new task to control where it appears in `cci task list`. Maybe a new "Data" group that includes this as well as the bulk data tasks?

File: cumulusci/tasks/salesforce/custom_settings.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
+import os
+import yaml
+
+from cumulusci.tasks.salesforce import BaseSalesforceApiTask
+from cumulusci.utils import os_friendly_path
+from cumulusci.core.exceptions import TaskOptionsError, CumulusCIException
+
+
+class UploadCustomSettings(BaseSalesforceApiTask):
```
Comment (davisagli) [inline cumulusci/tasks/salesforce/custom_settings.py]: Consider calling it LoadCustomSettings for better symmetry with the extract/load terminology we use in the bulk data tasks.

File: cumulusci/tasks/salesforce/custom_settings.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        -
+            location:
+                user:
+                    name: test@example.com
+            data:
+                MyField__c: 2"""
+
+    task_options = {
+        "settings_path": {
+            "description": "The path to a YAML settings store",
```
Comment (davisagli) [inline cumulusci/tasks/salesforce/custom_settings.py]:
```suggestion
            "description": "The path to a YAML settings file",
```

File: cumulusci/tasks/salesforce/custom_settings.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+                                    "email"
+                                ]
+                                query = f"SELECT Id FROM User WHERE Email = '{email_address}'"
+                        elif "org" == setting_instance["location"]:
+                            # Assign the Setup Owner Id to the organization.
+                            query = "SELECT Id FROM Organization"
+
+                    if query is None:
+                        raise CumulusCIException(
+                            f"No valid Setup Owner assignment found for Custom Setting {custom_setting}"
```
Comment (davisagli) [inline cumulusci/tasks/salesforce/custom_settings.py]: Can this be rewritten to give the user a clue about how they could resolve the problem?

File: cumulusci/tasks/salesforce/custom_settings.py [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+                    )
+
+                    setting_instance["data"].update({"SetupOwnerId": setup_owner_id})
+                    if existing_records["totalSize"] == 0:
+                        proxy_obj.create(setting_instance["data"])
+                    else:
+                        proxy_obj.update(
+                            existing_records["records"][0]["Id"],
+                            setting_instance["data"],
+                        )
```
Comment (davisagli) [inline cumulusci/tasks/salesforce/custom_settings.py]: Can you add logging of which settings objects are being updated in which location? I think that'll be useful for debugging.

File: cumulusci/tasks/salesforce/custom_settings.py [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+
+    def _load_settings(self):
+        # For each top-level heading in our YAML doc, create one or more
+        # custom settings.
+
+        for custom_setting, settings_data in self.settings.items():
+            proxy_obj = getattr(self.sf, custom_setting)
+            # If this level is a dict, we're working with a List Custom Setting
+            # If it's a list, we have a Hierarchy Custom Setting.
+            if isinstance(settings_data, dict):
```
Comment (prescod) [inline cumulusci/tasks/salesforce/custom_settings.py]: I would be inclined to have a method for each branch.

And an else clause that raises an error about the YAML format.

Comment (davidmreed) [reply cumulusci/tasks/salesforce/custom_settings.py]: I added another branch and a new test case to snag bad YAML.

**Pull request merged**